gcovr/tests/test_args.py

```diff
@@ -150,6 +159,53 @@ def test_non_existing_directory_2_csv(capsys):
     helper_test_non_existing_directory_2_output(capsys, '--csv')
 
 
+def helper_test_non_writable_directory_output(capsys, option):
+    c = capture(capsys, [option, '/file.txt'])
+    assert c.out == ''
+    assert 'Could not create output file \'/file.txt\': ' in c.err
+    assert c.exception.code != 0
+
+
+@pytest.mark.skipif(sys.platform == 'win32', reason="Not for Windows")
+def test_non_writable_directory_output(capsys):
+    helper_test_non_writable_directory_output(capsys, '--output')
+
+
+@pytest.mark.skipif(sys.platform == 'win32', reason="Not for Windows")
+def test_non_writable_directory_txt(capsys):
+    helper_test_non_writable_directory_output(capsys, '--txt')
```

Comment:
It seems like these tests would be the perfect use case for `@pytest.mark.parametrize()`, but I recall there were some problems last time we tried this.

Reply:
Yes, `parametrize` didn't work together with the capsys argument.